components/event-signup.js

```diff
@@ -0,0 +1,213 @@
+"use client";
+
+import { useForm } from "react-hook-form";
+import { useState, useEffect } from "react";
+
+function EventSignup(eventProps) {
+  const [eventData, setEventData] = useState();
+  const [registerSuccess, setRegisterSuccess] = useState();
+  const [ticketUrl, setTicketUrl] = useState("");
+
+  useEffect(() => {
+    checkSlug();
+  }, []);
+
+  const {
+    register,
+    handleSubmit,
+    formState: { errors, isSubmitSuccessful, isSubmitting },
+  } = useForm({
+    mode: "onTouched",
+  });
+
+  const checkSlug = async () => {
+    const eventData = fetch("/api/tito-check-slug")
+      .then((response) => response.json())
+      .then((json) => {
+        setEventData(json);
+      })
+      .catch((error) => {
+        res.json(error);
+        res.status(405).end();
+      });
+  };
+
+  const onSubmit = (data) => {
+    const registration = {
+      registrationData: {
+        email: data.email,
+        name: data.name,
+        discount_code: "",
+        source: "",
+        notify: true,
+        line_items: [{ release_id: data.attendance, quantity: 1 }],
+      },
+      nextEvent: eventData.nextEvent,
+    };
+
+    const createRegistration = fetch("/api/tito-create-registration", {
+      method: "POST",
+      body: JSON.stringify(registration),
+    })
+      .then((response) => response.json())
+      .then((json) => {
+        setRegisterSuccess(true);
+        setTicketUrl(json.ticket);
+      })
+      .catch((error) => {
+        Error.throw(
```

Comment:
Probably want to set this in an error state to render rather than throwing an error for some nextjs error boundary to pickup.

Reply:
Resolved in commit: c18ad30b5eefc35725dbdffc4510cd29641bddd4